### PR TITLE
enable autoscaler in cc-based cluster

### PR DESCRIPTION
**Pull request by zhouke1991**
### What this PR does / why we need it
Add autoscaler(optional) to the utkg cluster lifecycle.

### CODE REVIEW

**Comment by zhouke1991** [comment]
//usebom:sandbox/v1.7.0-zshippable/430690895461907853/tkg-compatibility:v12

**Comment by zhouke1991** [comment]
//usebom:sandbox/v1.7.0-zshippable/430854692847751580/tkg-compatibility:v12

**Review by wjun — APPROVED** [review]
LGTM.

File: tkg/clusterclient/clusterclient.go [hunk 1]
```diff
@@ -685,6 +691,47 @@ func (c *client) WaitForAutoscalerDeployment(deploymentName, namespace string) e
 	return c.GetResource(&appsv1.Deployment{}, deploymentName, namespace, VerifyAutoscalerDeploymentAvailable, &PollOptions{Interval: CheckResourceInterval, Timeout: CheckAutoscalerDeploymentTimeout})
 }
 
+func (c *client) ApplyPatchForAutoScalerDeployment(tkgBomClient tkgconfigbom.Client, clusterName string, k8sVersion string, namespace string) error {
+	var autoScalerDeployment appsv1.Deployment
+	autoscalerDeploymentName := clusterName + "-cluster-autoscaler"
```
Comment (wjun) [inline tkg/clusterclient/clusterclient.go:696]: Use constant value. This can be fixed in a later pr.

**Pull request merged**